packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -0,0 +1,93 @@
+import { Fn } from './cfn-fn';
+import { Aws } from './cfn-pseudo';
+
+/*
+ * This file contains functions for translating from a pure CFN value
+ * (like a JS object { "Ref": "Bucket" })
+ * to a form CDK understands
+ * (like Fn.ref('Bucket')).
+ */
+
+export function cfnValueToCdkValue(cfnValue: any): any {
+  if (cfnValue === undefined || cfnValue === null) {
+    return undefined;
+  }
+  if (typeof cfnValue === 'string' || typeof cfnValue === 'number') {
+    return cfnValue;
+  }
+  if (Array.isArray(cfnValue)) {
+    return cfnValue.map(el => cfnValueToCdkValue(el));
+  }
+  if (typeof cfnValue === 'object') {
+    // An object can be either a CFN intrinsic, or an actual object
+    const cfnIntrinsic = serializeIfCfnIntrinsic(cfnValue);
+    if (cfnIntrinsic) {
+      return cfnIntrinsic;
+    }
+    const ret: any = {};
+    for (const [key, val] of Object.entries(cfnValue)) {
+      ret[key] = cfnValueToCdkValue(val);
+    }
+    return ret;
+  }
+}
+
+function serializeIfCfnIntrinsic(object: any): any {
+  const key = looksLikeCfnIntrinsic(object);
+  switch (key) {
+    case undefined:
+      return undefined;
+    case 'Ref': {
+      const specialCaseRef = specialCaseRefs(object[key]);
```

Comment:
Hack!

```ts
return specialCaseRefs(object[key]) ?? Fn._ref(object[key]);
```

Reply:
I think "Hack!" might be a little strong here :) Done!